### ISSUE TITLE
Assets not found in development

### ISSUE THREAD

**Issue by arnoson** [andrefelipe/vite-php-setup#2]
I tested the setup but I can't get assets working in development mode when using a php dev server inside public. The vue logo is referenced as `/assets/logo.png`. So when running a php dev server under `localhost:8888` the browser tries to load the image as `localhost:8888/assets/logo.png` but it should look for `localhost:3000/assets/logo.png` . Of course I could also mirror all my assets from `src` to the `public` folder but this doesn't seem to be the right. Do you have found a workaround for this problem or are using another php server setup?

**Comment by andrefelipe** [comment]
Hello @arnoson 

Vite referencing the vue logo as ``/assets/logo.png`` is correct. And on production it will work correctly.

The solution is to create a symlink in your computer to map this ``/assets`` folder to your Vite's ``/src/assets``

Something like this is what your need:
```
ln -s {path_to_vite}/src/assets {path_to_public_html}/assets
```

It's a expected limitation according to Vite documentation. I tried of course to overcome this in many ways but couldn't. 

I will attach a screenshot here to illustrate my situation. I am using Laravel Valet as dev server.

![Screen Shot 2021-02-12 at 14 07 37](https://user-images.githubusercontent.com/308009/107799947-e60cdc80-6d3c-11eb-8044-179a4054e9dc.png)

**Comment by arnoson** [comment]
hey @andrefelipe, thanks for the quick reply!
this sounds like a good soution 👍 
However I can't get it working with  laravel valet, did you have to make some configurations or should valet follow symlinks by default? (maybe the problem is also related to WSL2 that I'm using)

**Comment by arnoson** [comment]
My bad, I tried to create the symlink with a relative path so it wasn't working,
Thank you!

**Comment by hubtraum** [comment]
@arnoson I wanted to share a thought on this topic since I stumbled upon this issue too. 
I added the following to my vite.config.js:

```
css: {
    preprocessorOptions: {
      scss: { additionalData: `$cdnUrl: "https://127.0.0.1:3001";` }, // replace 127.0.0.1:3001 with your dev server ip:port
    },
  },
```

and within my scss file:

`background-image: url("#{$cdnUrl}/assets/logo.png")`

I placed the file logo.png in `/public/assets/` in the project root dir.

This results in static assets being served correctly via the vite dev server, and no symlink is needed.

Your thoughts on this approach? My idea is, that one only thing needed further is to make a difference between dev/production for the $cdnUrl in my example:

```
css: {
      preprocessorOptions: {
        scss: { additionalData: mode == "development" ? `$cdnUrl: "https://127.0.0.1:3001";` : `$cdnUrl: "";` },
      },
    },
```
